pkg/controllers/work/patch_util.go

```diff
@@ -107,19 +108,12 @@ func setModifiedConfigurationAnnotation(obj runtime.Object) error {
 	}
 	// set the last applied annotation back
 	annotations[LastAppliedConfigAnnotation] = string(modified)
-	if err = metadataAccessor.SetAnnotations(obj, annotations); err != nil {
-		return err
-	}
-	annotations, err = metadataAccessor.Annotations(obj)
-	if err != nil {
-		return err
-	}
 	if err := validation.ValidateAnnotationsSize(annotations); err != nil {
-		klog.InfoS(fmt.Sprintf("setting last applied config annotation to empty, %s", err))
+		klog.V(2).InfoS(fmt.Sprintf("setting last applied config annotation to empty, %s", err))
 		annotations[LastAppliedConfigAnnotation] = ""
-		if err = metadataAccessor.SetAnnotations(obj, annotations); err != nil {
-			return err
-		}
+	}
+	if err = metadataAccessor.SetAnnotations(obj, annotations); err != nil {
+		return err
 	}
 	return nil
```

Comment:
```suggestion
	return metadataAccessor.SetAnnotations(obj, annotations)
```

to simplify the code